src/models/site/useSiteSettings.js

```diff
@@ -1,100 +1,63 @@
-import { useContext, useEffect, useState } from 'react';
-import { merge } from 'lodash-es';
+import { get, merge } from 'lodash-es';
 import axios from 'axios';
-import getAxiosResponse from '../../utils/getAxiosResponse';
-import getResponseVersion from '../../utils/getResponseVersion';
-import { formatError } from '../../utils/formatters';
-import {
-  AppContext,
-  setSiteSettingsNeedsFetch,
-  setSiteSettingsSchema,
-  setSiteSettingsVersion,
-  setSiteSettings,
-  setSiteSettingsSchemaNeedsFetch,
-} from '../../context';
+import { useQuery } from 'react-query';
 
-export default function useSiteSettings() {
-  const { state, dispatch } = useContext(AppContext);
-  const {
-    siteSettings,
-    siteSettingsSchema,
-    siteSettingsVersion,
-    siteSettingsNeedsFetch,
-    siteSettingsSchemaNeedsFetch,
-  } = state;
+import queryKeys from '../../constants/queryKeys';
 
-  const [error, setError] = useState(null);
-  const [settingsLoading, setSettingsLoading] = useState(
-    siteSettingsNeedsFetch,
-  );
-  const [schemaLoading, setSchemaLoading] = useState(
-    siteSettingsSchemaNeedsFetch,
+export default function useSiteSettings() {
+  const settingsSchemaResult = useQuery(
+    queryKeys.settingsSchema,
+    async () => {
+      const response = await axios({
+        url: `${__houston_url__}/api/v1/configurationDefinition/default/__bundle_setup`,
+        timeout: 2000,
+      });
+      return get(response, 'data.response.configuration');
+    },
+    {
+      staleTime: Infinity,
```

Comment:
?

Reply:
you'll have to take a look at the `react-query` docs for this one